File: src/javascript/compress_ew5_.js
const assert = require("assert");

ABC = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
compress = (s, k = [...s], c = 1 + k.join(1)) => {
  return [...ABC].reduce((_p, a) => {
    k.reduce(
      (p1, v1, i) =>
        k.reduce((p2, v2, j) =>
          // console.log(i,j,c)&&
          // i = j = 1;
          c.replaceAll(i + a + j + a, i + j + a)
        ),
      c
    );
  }, c);
};

assert.equal(compress("AAABBAAC"), "3A2B2A1C");

File: src/python/compress_ew5.py
from functools import reduce

#      v--- 114 bytes 
compress=lambda s:reduce(lambda p,v:(p+v,p[:-2]+str(int(v[0])+int(p[-2]))+str(p[-1]))[p[-1]==v[1]],map(lambda v:"1"+v,s))

assert compress("AAABBAAC") == "3A2B2A1C"
